pkg/controllers/generic/syncer.go

```diff
@@ -0,0 +1,309 @@
+package generic
```

Comment:
Would be good to have a test for this for single namespace as well as multi namespace mode. Preferrably a unit test